packages/devtools_server/lib/src/server.dart

```diff
@@ -52,23 +53,25 @@ final argParser = ArgParser()
     help: 'Launches DevTools in a browser immediately at start.',
   );
 
-void serveDevToolsWithArgs(List<String> arguments) async {
+Future<HttpServer> serveDevToolsWithArgs(List<String> arguments) async {
   final args = argParser.parse(arguments);
 
   final help = args[argHelp];
   final bool machineMode = args[argMachine];
   final bool launchBrowser = args[argLaunchBrowser];
   final port = args[argPort] != null ? int.tryParse(args[argPort]) ?? 0 : 0;
 
-  serveDevTools(
+  return serveDevTools(
       help: help,
       machineMode: machineMode,
       launchBrowser: launchBrowser,
       port: port);
 }
 
-void serveDevTools({
+Future<HttpServer> serveDevTools({
   bool help = false,
+  bool enablePrinting = true,
```

Comment:
maybe make an enum of
`{silent, stdout, machineMode}`
as 
`enablePrinting = false, machineMode=true` would be broken?
alternately add an assert that machineMode is never true at the same time enablePrinting=false

I'm also fine with changing `enablePrinting` to `!silent`.